File: backend/src/main/scala/com/skewwhiffy/auraltester/notes/DirectedInterval.scala
package com.skewwhiffy.auraltester.notes

import com.skewwhiffy.auraltester.notes.Interval.Interval

class DirectedInterval(
  val interval: Interval,
  val direction: IntervalDirection
) {
  override def equals(obj: Any): Boolean = obj match {
    case other: DirectedInterval => other.interval == interval && other.direction == direction
    case _ => false
  }
}

File: backend/src/main/kotlin/com/skewwhiffy/auraltester/notation/model/interval/DirectedInterval.kt
package com.skewwhiffy.auraltester.notation.model.interval

data class DirectedInterval(val interval: Interval, val direction: IntervalDirection) {
    operator fun unaryMinus(): DirectedInterval = when (direction) {
        IntervalDirection.DOWN -> DirectedInterval(interval, IntervalDirection.UP)
        IntervalDirection.UP -> DirectedInterval(interval, IntervalDirection.DOWN)
    }
}
